drivers/soundwire/intel.c

```diff
@@ -449,7 +449,7 @@ static void intel_shim_wake(struct sdw_intel *sdw, bool wake_enable)
 		/* Clear wake status */
 		wake_sts = intel_readw(shim, SDW_SHIM_WAKESTS);
 		wake_sts |= (SDW_SHIM_WAKEEN_ENABLE << link_id);
-		intel_writew(shim, SDW_SHIM_WAKESTS_STATUS, wake_sts);
+		intel_writew(shim, SDW_SHIM_WAKESTS, (wake_sts & SDW_SHIM_WAKESTS_STATUS));
```

Comment:
the register change is fine, I don't see why you added an AND operation that will kill what is done in line 451. Something's not right here.

Reply:
@plbossart I referred the ACE1.x spec. It says: ``` (SNDWSC-1)*(1-SNDWMOD):0 : PREQ/WakeUp Status```  and  ```15:(SNDWSC-1)*(1-SNDWMOD)+1 : Reserved SW must use zeros for writes``` The cAVS2.x has the similar description. So I believe ```wake_sts & SDW_SHIM_WAKESTS_STATUS``` is the right writing. The only problem is how to set the ```SDW_SHIM_WAKESTS_STATUS```. For the ACE1.x and cAVS2.x, both are 0xf. And I'm still checking other platforms' setting (I don't have the spec so far). What about after I have checked other platforms setting, I submit another patch for ```SDW_SHIM_WAKESTS_STATUS``` setting?

Reply:
@libinyang please don't refer embargoed codenames in public